Makefile

```diff
@@ -56,70 +56,68 @@ build_compiled:
 		--target $(COMPILE_TARGET) .
 
 build_serialize:
-	BUILD_ARGS="$(BUILD_ARGS) --build-arg serialize=true" COMPILED_IMAGE=$(SERIALIZE_IMAGE) $(MAKE) build_compiled
+	BUILD_ARGS="$(BUILD_ARGS) --build-arg serialize=true" \
+	COMPILED_IMAGE=$(SERIALIZE_IMAGE)-check-md5 \
```

Comment:
If you are undoing the association, this and the other places this is referred to would be removed. e.g. build_serialize being a general endpoint for compiling the model with serialization on then making a very specific tag is a little confusing. But, it is not a big deal, and we may end up changing the naming conventions back to your original proposal anyways